mlrun/runtimes/function.py

```diff
@@ -167,6 +168,114 @@ def add_trigger(self, name, spec):
         self.spec.config[f"spec.triggers.{name}"] = spec
         return self
 
+    def from_remote_source(
+        self, source, handler="", runtime="", credentials=None,
+    ):
+        """
+        Load nuclio function from remote source
+        :param source: a full path to the nuclio function source (code entry) to load the function from
+        :param handler: a path to the function's handler, including path inside archive/git repo
+        :param runtime: (optional) the runtime of the function (defaults to python:3.7)
+        :param credentials: a dictionary of credentials to be used to fetch the function from the source, options:
+         ["V3IO_ACCESS_KEY",
+         "GIT_USERNAME",
+         "GIT_PASSWORD",
+         "AWS_ACCESS_KEY_ID",
+         "AWS_SECRET_ACCESS_KEY",
+         "AWS_SESSION_TOKEN"]
+
+        Examples:
+        git:
+        ("git://github.com/org/repo#my-branch",
+         handler="path/inside/repo#main:handler",
+         credentials={"GIT_PASSWORD": "my-access-token"})
+        s3:
+        ("s3://my-bucket/path/in/bucket/my-functions-archive",
+         handler="path/inside/functions/archive#main:Handler",
+         runtime="golang",
+         credentials={"AWS_ACCESS_KEY_ID": "some-id", "AWS_SECRET_ACCESS_KEY": "some-secret"})
+        """
```

Comment:
@sahare92 i dont think the doc string will render ok (e.g. code examples should be idented and after "::", u should check it